active_directory_ldap/komand_active_directory_ldap/util/utils.py

```diff
@@ -144,3 +148,66 @@ def escape_brackets_for_query(query: str, pairs: dict) -> str:
                 query = query[:value] + "\\29" + query[value + 1 :]
                 query = query[:key] + "\\28" + query[key + 1 :]
         return query
+
+    @staticmethod
+    def escape_user_dn(user_dn: str) -> str:
+        pairs = ADUtils.find_parentheses_pairs(user_dn)
+        if pairs:
+            # replace ( and ) when they are part of a name rather than a search parameter
+            user_dn = ADUtils.escape_brackets_for_query(user_dn, pairs)
+
+        return user_dn
+
+    @staticmethod
+    def check_user_dn_is_valid(conn, user_dn: str, search_base: str) -> bool:
+        try:
+            conn.search(
+                search_base=search_base,
+                search_filter=f'(distinguishedName={ADUtils.escape_user_dn(user_dn)})',
+                attributes=['userAccountControl']
+            )
+        except LDAPInvalidDnError as e:
+            raise PluginException(
+                cause='The server not found group.',
+                assistance='Check group name.',
+                data=e
+            )
+        except LDAPOperationsErrorResult as e:
+            raise PluginException(
+                preset=PluginException.Preset.SERVER_ERROR,
+                data=e
+            )
+        return len([d['dn'] for d in conn.response if 'dn' in d]) > 0
+
+    @staticmethod
+    def change_account_status(conn, dn: str, status: bool, logger: Logger) -> bool:
+        dn, search_base = ADUtils.format_dn(dn)
+        logger.info(f'Escaped DN {dn}')
+
+        if not ADUtils.check_user_dn_is_valid(conn, dn, search_base):
+            logger.error(f'The DN {dn} was not found')
+            raise PluginException(
+                cause='The DN was not found.',
+                assistance=f'The DN {dn} was not found.'
+            )
+        user_list = [d["attributes"] for d in conn.response if "attributes" in d]
+        user_control = user_list[0]
+        try:
+            account_status = user_control['userAccountControl']
+        except Exception as ex:
+            logger.error('The DN ' + dn + ' is not a user')
+            raise PluginException(cause='The DN is not a user',
```

Comment:
Computers in AD are also users (with $ at the end of the sAMAccountName) and have userAccountControl attributes.  How about changing the message to "The DN is not a user object" and assistance to "The DN ' + dn + ' is not a user object and therefore can't be enabled or disabled"
It's not perfect - since you can create user and computer objects without a userAccountControl attribute - but it's good enough for 99.9% of cases.

Reply:
For this one, how about

```
raise PluginException(cause=f'The DN {dn} is not a user object therefore it cannot be enabled or disabled',
                              assistance='Please provide a valid user object and try again.') from ex
```